src/coreclr/vm/dacenumerablehash.inl

```diff
@@ -129,21 +130,24 @@ void DacEnumerableHashTable<DAC_ENUM_HASH_ARGS>::BaseInsertEntry(DacEnumerableHa
     // Remember the entry hash code.
     pVolatileEntry->m_iHashValue = iHash;
 
-    // Compute which bucket the entry belongs in based on the hash.
-    DWORD dwBucket = iHash % m_cBuckets;
+    auto curBuckets = GetBuckets();
+    DWORD cBuckets = (DWORD)((size_t*)curBuckets)[0];
```

Comment:
This cast won't work in the DAC. It will be indexing into local memory addresses and is a dangerous operation. dac_cast is probably what you want, and you'll need to get the PTR_VolatileEntry and do a dac_cast of that and read an int as needed..

Reply:
So far I've managed to avoid dealing with DPTR much. I guess it's a good time to start.